### PR TITLE
Set_get_bparam_changes

### PR DESCRIPTION
**Pull request by Arceveti**

### CODE REVIEW

File: include/macros.h [hunk 1]
```diff
@@ -71,6 +71,8 @@
 
 #define MODE_NTSC 0
 #define MODE_MPAL 1
-#define MODE_PAL 2
+#define MODE_PAL  2
+
+#define FORCE_CRASH { *(vs8*)0 = 0; }
```
Comment (thecozies) [inline include/macros.h:76]: why'd you move this here? types.h is explicitly included _everywhere_ so it'd be easier to access there no?

File: include/object_constants.h [hunk 2]
```diff
@@ -127,42 +127,85 @@ enum DialogState {
...
+#define GET_BPARAM12(behParams) GET_BPARAMS((behParams), 1, 2)
+#define GET_BPARAM23(behParams) GET_BPARAMS((behParams), 2, 2)
+#define GET_BPARAM34(behParams) GET_BPARAMS((behParams), 3, 2)
+
+// Read 3 bparams as a single value
+#define GET_BPARAM123(behParams) GET_BPARAMS((behParams), 1, 3)
+#define GET_BPARAM234(behParams) GET_BPARAMS((behParams), 2, 3)
+
+// Read 4 bparams as a single value
+#define GET_BPARAM1234(behParams) (behParams)
```
[the remark this answers is not in the document]
Comment (Arceveti) [reply include/object_constants.h:172]: True, I just put it there for consistency lol

File: include/object_constants.h [hunk 3]
```diff
@@ -1402,19 +1445,21 @@ enum BehParam1StarAct { // BPARAM1
...
-    #define GOOMBA_SIZE_TINY                                0x2
-    #define GOOMBA_BP_SIZE_MASK                             0x3
-    #define GOOMBA_BP_TRIPLET_FLAG_MASK                     0x000000FC
+    #define GOOMBA_SIZE_REGULAR                             0x00
+    #define GOOMBA_SIZE_HUGE                                0x01
+    #define GOOMBA_SIZE_TINY                                0x02
+    #define GOOMBA_BP_SIZE_MASK                             0x03
+    #define GOOMBA_BP_TRIPLET_FLAG_MASK                     0xFC
+    /* BPARAM3 */
+    #define GOOMBA_BP3_IS_FLOOMBA                           (1 << 7)
```
Comment (thecozies) [inline include/object_constants.h]: this is gonna collide with your other PR right? so the goomba one is going to need to wait on this?

Comment (Arceveti) [reply include/object_constants.h]: It would have but I fixed it just now.

Comment (thecozies) [reply include/object_constants.h]: @Arceveti wait for the parent to get merged, put a comment saying something like "depends on PR 80", that way there isn't anything broken

Comment (thecozies) [reply include/object_constants.h]: its fine this time

File: include/object_constants.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (thecozies) [reply include/object_constants.h:172]: yeahhhhh but its soo useless i'd rather it just not exist haha

**Comment by thecozies** [comment]
@Arceveti i didnt

approve

**Comment by Arceveti** [comment]
Oh huh it said you did on my end

File: include/object_constants.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (thecozies) [reply include/object_constants.h:172]: @Arceveti revert this pls

[the remark this answers is not in the document]
Comment (Arceveti) [reply include/object_constants.h:172]: Removed

[the remark this answers is not in the document]
Comment (thecozies) [reply include/object_constants.h:172]: where?

[the remark this answers is not in the document]
Comment (Arceveti) [reply include/object_constants.h:172]: tfw pushed to wrong branch

**Pull request merged**